luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/LuthierIndustrialPark.java

```diff
@@ -159,15 +168,56 @@ public DataSourceMetadataService getMetadataService() {
      *
      * @return the LogicalMetric instance corresponding to this metricName.
      */
-    // TODO: to be finished in the metric PR.
     public LogicalMetric getMetric(String metricName) {
+        // TODO: to be finished in the metric PR.
         return null;
     }
 
+    /**
+     * Retrieve or build a LogicalTable.
+     *
+     * @param tableId the identifier for table name and granularity to be retrieved or built
+     *
+     * @return the logicalTable instance corresponding to this TableIdentifier.
+     */
+    public LogicalTable getLogicalTable(TableIdentifier tableId) {
+        LogicalTableDictionary logicalTableDictionary = resourceDictionaries.getLogicalDictionary();
+        if (! logicalTableDictionary.containsKey(tableId)) {
+            String tableName = tableId.getKey() + tableId.getValue().toString();
+            LogicalTableGroup logicalTableGroup = logicalTableGroupFactoryPark.buildEntity(tableName, this);
+            logicalTableDictionary.putAll(logicalTableGroup);
+        }
+        return logicalTableDictionary.get(tableId);
+    }
+
+    /**
+     * Retrieve or build a LogicalTable.
+     *
+     * @param tableName  the name of the logical table to be retrieved or built
+     * @param granularity  the granularity of the logical table to be retrieved or built
+     *
+     * @return the logicalTable instance corresponding to this name and granularity.
+     */
+    public LogicalTable getLogicalTable(String tableName, Granularity granularity) {
+        return getLogicalTable(new TableIdentifier(tableName, granularity));
+    }
+
+    /**
+     * Private setters to put a group of LogicalTable's.
+     *
+     * @param tableKey named tableKey according to the definition in the json Config
+     */
+    private void putLogicalTableByGroup(String tableKey) {
+        LogicalTableDictionary logicalTableDictionary = resourceDictionaries.getLogicalDictionary();
+        LogicalTableGroup logicalTableGroup = logicalTableGroupFactoryPark.buildEntity(tableKey, this);
+        logicalTableDictionary.putAll(logicalTableGroup);
+    }
+
     @Override
     public void load() {
         dimensionFactoryPark.fetchConfig().fieldNames().forEachRemaining(this::getDimension);
         physicalTableFactoryPark.fetchConfig().fieldNames().forEachRemaining(this::getPhysicalTable);
+        logicalTableGroupFactoryPark.fetchConfig().fieldNames().forEachRemaining(this::putLogicalTableByGroup);
```

Comment:
This should render everything else in this method redundant. We should try running things with this the only
statement in `load` and see if it works.

Reply:
That should work in production but not in all of the unit tests. 
The reason is, in unit tests, we directly call LIP.builder.build(), which checks the granularityDictionary, sees that's `null`, and then skips the load of logicalTables, thus the recursive process will not happen. 

On the other hand, if we go through the LuthierBinderFactory to initialize LIP, we are good. 
If the test directly get a specific concept, then the recursive process start from that point, which also works as expected.

Reply:
I'm not sure I understand the reasoning.  First, granularity dictionary should never default to null, so that failout path shouldn't happen.

Out unit test cases should be 'test build...' and 'test getFoo(FooName)' , right?

Reply:
Basically. In fact, you should be able to get away with just `test getFoo(FooName)` because if `FooName` doesn't exist, the system should automatically build it and all its dependencies.